Release: 5.9.0

adrianostanley opened the pull request:
# Summary

* Bumps FW to 5.9.0 / namespaces to 5_9_0

## Notes

I was following the [Deploying new versions](https://github.com/skyverge/wc-plugin-framework/wiki/Deploying-new-versions) article but since this is the first time, here are a couple of notes to be known before merging:

* My test environment is not 100% reliable at this point and I was able to assert only the **unit** and **integration** ones. Please, run the acceptance tests before merging.
* While running `grunt build` I had a few warnings:

```
Running "uglify:uglify" (uglify) task
>> Destination woocommerce/payment-gateway/assets/js/frontend/sv-wc-payment-gateway-frontend.min.js not written because src files were empty.
>> No files created.
```

and

```
Running "sass:compile" (sass) task
Warning: 
You need to have Ruby and Sass installed and in your PATH for this task to work.
```

* But the script finished and the POT file was generated.
* In case I am not here when this PR is reviewed, here are the changelog entries to be copied to the [release](https://github.com/skyverge/wc-plugin-framework/releases/new) description:

```
 * Feature - Add helper method to detect whether the current request is a REST API request
 * Fix - Fix a Capture Charge button issue when adding items to an order
 * Fix - Harden code to avoid a potential PHP warning at checkout if no card logos are configured for a gateway
 * Fix - Update Pre-Orders and Subscriptions integrations to use 2 digits expiration years
```

## Before merge

- [x] Acceptance tests pass
- [x] I have confirmed these changes in each supported minor WooCommerce version
- [x] All required plugin changes have been documented for this new version

wvega reviewed (approved):
@adrianostanley the changes look good! Acceptance tests are passing locally for me, and all tests are passing on Travis CI.

Only one change: I replaced a couple of `x.y.z-dev` versions with `5.9.0`: https://github.com/skyverge/wc-plugin-framework/pull/506/commits/b69069460ccc3db80342db0bdbd7c0390c209397

>Running "uglify:uglify" (uglify) task
>Destination woocommerce/payment-gateway/assets/js/frontend/sv-wc-payment-gateway-frontend.min.js not written because src files were empty.
>No files created.

I think that's fine. That file doesn't seem to be in the project anymore, so the error seems expected to me.

>Running "sass:compile" (sass) task
>Warning: 
>You need to have Ruby and Sass installed and in your PATH for this task to work.

I'm not sure about this one. I think I saw it at some point, but I can't remember what was the solution for me. Maybe try `npm install -g sass` from https://github.com/gruntjs/grunt-contrib-sass/issues/229#issuecomment-631360543.

adrianostanley commented:
> @adrianostanley the changes look good! Acceptance tests are passing locally for me, and all tests are passing on Travis CI.
> 
> Only one change: I replaced a couple of `x.y.z-dev` versions with `5.9.0`: [b690694](https://github.com/skyverge/wc-plugin-framework/commit/b69069460ccc3db80342db0bdbd7c0390c209397)

@wvega sounds good! Thanks for that. I went ahead and updated the [https://github.com/skyverge/wc-plugin-framework/wiki/Deploying-new-versions](https://github.com/skyverge/wc-plugin-framework/wiki/Deploying-new-versions) with another step: _Do a search for `x.y.z` and replace them for the new version number_. If you think that's not a general rule, let me know and I'll rollback those changes from the wiki.

> 
> > Running "uglify:uglify" (uglify) task
> > Destination woocommerce/payment-gateway/assets/js/frontend/sv-wc-payment-gateway-frontend.min.js not written because src files were empty.
> > No files created.
> 
> I think that's fine. That file doesn't seem to be in the project anymore, so the error seems expected to me.
> 
> > Running "sass:compile" (sass) task
> > Warning:
> > You need to have Ruby and Sass installed and in your PATH for this task to work.
> 
> I'm not sure about this one. I think I saw it at some point, but I can't remember what was the solution for me. Maybe try `npm install -g sass` from [gruntjs/grunt-contrib-sass#229 (comment)](https://github.com/gruntjs/grunt-contrib-sass/issues/229#issuecomment-631360543).

`npm install -g sass` went well and `grunt build` didn't end up with warnings. Since they didn't generate new changes nor commits, I'll proceed and merge this PR. Thanks for reviewing.

wvega commented:
>I went ahead and updated the https://github.com/skyverge/wc-plugin-framework/wiki/Deploying-new-versions with another step: Do a search for x.y.z and replace them for the new version number.

@adrianostanley adding that step is a good idea 🎉 Thank you for updating the wiki!
